editor/editor_file_system.cpp

```diff
@@ -2006,6 +2004,36 @@ void EditorFileSystem::_update_script_documentation() {
 	update_script_paths_documentation.clear();
 }
 
+bool EditorFileSystem::_should_reload_script(const String &p_path) {
+	if (first_scan) {
+		return false;
+	}
+
+	Ref<Resource> res = ResourceCache::get_ref(p_path);
+	if (res.is_null()) {
+		// Script not already loaded.
+		return false;
+	}
+
+	Ref<Script> scr = res;
+	if (scr.is_null()) {
+		// Not a script.
+		return false;
+	}
+
+	// Scripts are reloaded via the script editor if they are currently opened.
+	if (ScriptEditor::get_singleton()->get_open_scripts().has(scr)) {
+		return false;
+	}
+
+	if (res->get_last_modified_time() == FileAccess::get_modified_time(p_path)) {
+		// Already up to date.
+		return false;
+	}
```

Comment:
I wonder if this can cause issues for the Android editor like #94416.
If the implementation for `get_modified_time` is still buggy on Android, this might be problematic. https://github.com/godotengine/godot/issues/94416#issuecomment-2267172619

Reply:
You are so right. I'm trying to find a way to prevent script double/triple script reloading from file.

After some more testing, I removed the `reloading_from_file` completely for script in `EditorNode`. I did not find a way to scripts not being reloaded from file now that EditorFileSystem do it in `EditorFileSystem::_update_script_documentation`. That way, no need to check the last modified time and not hack in `EditorNode` for scripts.

Technically, the method `EditorFileSystem::_update_script_documentation` should be renamed now for  `EditorFileSystem::_update_scripts` but I'm not sure it's a good idea to rename it in this PR. What do you think?